torch/hub.py

```diff
@@ -120,28 +121,57 @@ def _read_url(url):
 
 
 def _validate_not_a_forked_repo(repo_owner, repo_name, branch):
+    # Here we try to make sure that the branch/commit hash effectively belongs
+    # to the repo_owner/repo_name repo.
+    # This is important because GitHub is weird, and the download URL
+    # f'https://github.com/{repo_owner}/{repo_name}/archive/{commit_hash}.zip'
+    # may actually exist *even if* commit_hash is from a different repo_owner / fork.
+    # For example this URL exists:
+    # https://github.com/pytorch/vision/archive/8949c7011facf7801fdf077cc3e4ecd8f0940c7e.zip
+    # even though the commit hash doesn't come from pytorch/vision. It actually comes from a fork
+    # https://github.com/NicolasHug/vision/commit/8949c7011facf7801fdf077cc3e4ecd8f0940c7e
+    # An interesting example of this is https://news.ycombinator.com/item?id=24882921.
```

Comment:
nit: I personally found this story interesting and informative on the perspective of how GitHub works behind the scene, but this particular instance was a troll made around a serious law suite, so I would not put it in here, just to be safe. I think the goal of communicating the potential security issue is achieved even without it.